distribution/lib/Standard/Table/0.0.0-dev/src/Data/Type/Value_Type_Helpers.enso

```diff
@@ -75,3 +78,95 @@ find_common_type types strict =
         # Double check if Mixed was really allowed to come out.
         if types.contains Value_Type.Mixed then Value_Type.Mixed else
             Nothing
+
+type Operation_Type_Helpers
+    ## PRIVATE
+
+       Arguments:
+       - is_column: a function that checks if the provided value is a column.
+    Instance (is_column : Any -> Boolean)
+
+    ## PRIVATE
+       Finds the type of an argument to a column operation.
+
+       If the argument is a column, the type of that column is returned. If it
+       is an Enso value, the smallest `Value_Type` that can fit that value will
+       be returned (but the Database is free to widen it to the closest type
+       that it supports without warning).
+
+       Since there is no special type for `Nothing` and `Nothing` technically
+       can fit any nullable type, it usually needs to be handled specially. This
+       method returns `Nothing` if the value is `Nothing` - so the caller can
+       try to treat this value as fitting any type, or accordingly to specific
+       semantics of each method.
+    find_argument_type : Any -> Value_Type | Nothing
+    find_argument_type self value = case value of
+        Nothing -> Nothing
+        _ -> if self.is_column value then value.value_type else
+            Enso_Types.most_specific_value_type value use_smallest=True
+
+    ## PRIVATE
+       A helper that can be used to check the argument value type, while allowing
+       `Nothing` values too.
+
+       > Example
+
+         type_helpers.check_argument_type other Value_Type.expect_text <| operation
+    check_argument_type self value type_check ~action =
```

Comment:
Could you add a type signature here?

Reply:
Also it might be good to mention that `type_check` is an assertion that can throw an error, rather than a predicate.